distributed/scheduler.py

```diff
@@ -766,6 +766,58 @@ def ncores(self):
         return self._nthreads
 
 
+@final
+@cclass
+class Computation:
+    """
+    Collection tracking a single compute or persist call
+
+    See also
+    --------
+    TaskPrefix
+    TaskGroup
+    TaskState
+    """
+
+    _start: double
+    _groups: set
+    _code: List[str]
+    _recent: "Computation" = None
+
+    def __init__(self):
+        self._start = time()
+        self._groups = set()
+        self._code = []
+        Computation.recent = self
+
+    @property
+    def code(self):
+        return self._code
+
+    @property
+    def start(self):
+        return self._start
+
+    @property
+    def stop(self):
+        return max(tg.stop for tg in self.groups)
+
+    @property
+    def states(self):
+        tg: TaskGroup
+        return merge_with(sum, [tg._states for tg in self._groups])
+
+    @property
+    def groups(self):
+        return self._groups
+
+    def __repr__(self):
+        return (
+            f"Computation: {format_time(time() - self._start)} ago with {len(self.groups)} groups\n\n"
+            + "\n---------------------------\n".join(self.code)
```

Comment:
I don't think the plain repr should be that long. The code could be thousands of lines long and the repr is something we might see in a console, an IDE, a log message, CI, etc.

Reply:
Yeah, I don't disagree.  This was useful during debugging / writing this PR.  I'm happy to strip out the code.  We could also consider only collecting +-50 lines of code around the relevant line.